bodies/views.py

```diff
@@ -88,6 +90,22 @@ def follow(self, request, pk):
 
         return Response(status=204)
 
+    def get_events(self, request, pk):
+        """Get all events from pk uuid or strid.
+        {?archived} is optional arguement to fetch old events"""
+        body = self.get_body(pk)
+        # Get query param
+        archived = request.GET.get("archived")
+        if archived is None:
+            archived = '0'
+
+        if archived not in ('0', '1'):
+            return Response({"message": "Invalid archived parameter"}, status=400)
+
+        queryset = Event.objects.filter(bodies=body, archived=archived).order_by('-start_time')
```

Comment:
The archived flag is actually an unfortunate result of some of my misconceptions about postgres performance. We are not, and shouldn't be using it at all, so you can just remove that filter. Instead, what this should be doing is get events in the last 1 year if the archived flag is not present, and all events if present.